source/operations/monitoring/metrics-and-alerts.rst

```diff
@@ -0,0 +1,396 @@
+
+.. metric:: minio_heal_objects_heal_total
+
+   Objects healed in current self healing run
+
+.. metric:: minio_heal_objects_total
+
+   Objects scanned in current self healing run
+
+.. metric:: minio_heal_time_last_activity_nano_seconds
+
+   Time elapsed (in nano seconds) since last self healing activity. This is set
+   to -1 until initial self heal
+
+Scanner Metrics
+~~~~~~~~~~~~~~~
+
+.. metric:: minio_node_scanner_bucket_scans_finished
+
+   Total number of bucket scans finished since server start.
+
+.. metric:: minio_node_scanner_bucket_scans_started
+
+   Total number of bucket scans started since server start.
+
+.. metric:: minio_node_scanner_directories_scanned
+
+   Total number of directories scanned since server start.
+
+.. metric:: minio_node_scanner_objects_scanned
+
+   Total number of unique objects scanned since server start.
+
+.. metric:: minio_node_scanner_versions_scanned
+
+   Total number of object versions scanned since server start.
+
+.. metric:: minio_node_syscall_read_total
+
+   Total read SysCalls to the kernel. ``/proc/[pid]/io syscr``
+
+.. metric:: minio_node_syscall_write_total
+
+   Total write SysCalls to the kernel. ``/proc/[pid]/io syscw``
+
+S3 Metrics
+~~~~~~~~~~
+
+.. metric:: minio_bucket_traffic_sent_bytes
+
+   Total number of bytes of S3 traffic sent per bucket
+
+.. metric:: minio_bucket_traffic_received_bytes
+
+   Total number of bytes of S3 traffic received per bucket
+
+.. metric:: minio_s3_requests_error_total
+
+   Total number S3 requests with errors
+
+.. metric:: minio_s3_requests_inflight_total
+
+   Total number of S3 requests currently in flight
+
+.. metric:: minio_s3_requests_total
+
+   Total number S3 requests
+
+.. metric:: minio_s3_time_ttfb_seconds_distribution
+
+   Distribution of the time to first byte across API calls.
+
+.. metric:: minio_s3_traffic_received_bytes
+
+   Total number of s3 bytes received.
+
+.. metric:: minio_s3_traffic_sent_bytes
+
+   Total number of s3 bytes sent
+
+.. metric:: minio_s3_requests_errors_total
+
+   Total number of S3 requests with 4xx and 5xx errors
+
+.. metric:: minio_s3_requests_4xx_errors_total
+
+   Total number of S3 requests with 4xx errors
+
+.. metric:: minio_s3_requests_5xx_errors_total
+
+   Total number of S3 requests with 5xx errors.
+
+Internal Metrics
+~~~~~~~~~~~~~~~~
+
+.. metric:: minio_inter_node_traffic_received_bytes
+
+   Total number of bytes received from other peer nodes.
+
+.. metric:: minio_inter_node_traffic_sent_bytes
+
+   Total number of bytes sent to the other peer nodes.
+
+.. metric:: minio_node_file_descriptor_limit_total
+
+   Limit on total number of open file descriptors for the MinIO Server process.
+
+.. metric:: minio_node_file_descriptor_open_total
+
+   Total number of open file descriptors by the MinIO Server process.
+
+.. metric:: minio_node_io_rchar_bytes
+
+   Total bytes read by the process from the underlying storage system including
```

Comment:
Total **number of** bytes ...

Reply:
All metrics in this section need this for consistency with other metrics.

Reply:
Done